test/models/user_test.rb

```diff
@@ -70,4 +70,7 @@ def setup
     @user.save
     assert_equal mixed_case_email.downcase, @user.reload.email
   end
+  test "authenticated? should return false for a user with nil digest" do
+    assert_not @user.authenticated?('')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.